x-pack/auditbeat/module/system/process/process.go

```diff
@@ -252,6 +254,26 @@ func (ms *MetricSet) getProcessInfos() ([]*ProcessInfo, error) {
 
 		pInfo, err := process.Info()
 		if err != nil {
+			if os.Geteuid() != 0 {
+				if runtime.GOOS == "darwin" {
+					/*
+						Skip process info errors on darwin when not root -
+						these will usually be permission issues when trying to read
+						other user's processes. Unfortunately, os.IsPermission does not
+						work because it is a custom error created using errors.New in
+						getProcTaskAllInfo() in go-sysinfo/providers/darwin/process_darwin_amd64.go
+
+						TODO: Fix go-sysinfo to have better error.
+					*/
+					continue
+				}
+
+				if os.IsPermission(err) {
+					// Running as non-root, so we have no access to other user's processes.
```

Comment:
We should log a warning the first time this happens to let them know that there are processes that we cannot report on due to permissions. It will probably be useful if someone says “it’s not working”.